client-internal/src/main/java/org/terracotta/angela/client/util/IgniteClientHelper.java

```diff
@@ -113,13 +114,16 @@ private static void checkAgentHealth(Ignite ignite, String hostname, int igniteP
             " but the expected version is [" + AngelaVersion.getAngelaVersion() + "]");
       }
     } catch (IgniteException e) {
-      e.printStackTrace();
       throw new IllegalStateException("Node with name '" + nodeName + "' not found in the cluster", e);
     }
   }
 
   private static String getNodeName(String nodeName, int ignitePort) {
-    return nodeName + ":" + ignitePort;
+    try {
+      return (IpUtils.isLocal(nodeName) ? IpUtils.getHostName() : nodeName) + ":" + ignitePort;
+    } catch (RuntimeException e) {
+      return nodeName + ":" + ignitePort;
+    }
```

Comment:
`nodeName` parameter can come from many places, including sourced config file for nodes. 
So replaces each ref to localhost to the current hostname.
Also (and I don't know why), client arrays is defined with a hostname, but this hostname is not passed here. This is the instanceId that is passed... Why ? I don't know...